adapters/repos/db/vector/flat/index.go

```diff
@@ -289,7 +298,12 @@ func float32SliceFromByteSlice(vector []byte, slice []float32) []float32 {
 
 func (index *flat) Add(id uint64, vector []float32) error {
 	index.trackDimensionsOnce.Do(func() {
-		atomic.StoreInt32(&index.dims, int32(len(vector)))
+		size := int32(len(vector))
+		atomic.StoreInt32(&index.dims, size)
+		err := index.setDimensions(size)
```

Comment:
validation of persisted dimension may be needed otherwise after a restart it may overwrite previous dimension?

Reply:
ah, index.dims is calculated when the flat index is initialized

Reply:
@jeroiraz, `Add` can't be used until the constructor `New` returns which will set the `sync.Once` `trackDimensionsOnce` if it reads a dimension from disk.